omigami/feature_selector.py

```diff
@@ -69,55 +69,121 @@ def fit(self, X: NumpyArray, y: NumpyArray, groups: NumpyArray = None):
 
         # outer_loop = self._make_outer_loop(input_data)
         # self._results = self._execute_repetitions(outer_loop)
-        # self.selected_features = self.post_processor.select_features(self._results)
-        # self.is_fit = True
+        self.selected_features = self.post_processor.select_features(repetition_results)
+        self.is_fit = True
         return self
 
+    def get_groups(self, groups, size):
+        if groups is None:
+            logging.info("groups is not specified: i.i.d. samples assumed")
+            groups = np.arange(size)
+        return groups
+
     def _run_outer_loop(self, input_data, data_splitter, outer_split):
-        # TODO: refactor this class as needed
-        rfe = RecursiveFeatureEliminator(input_data.n_features)
+        rfe = RecursiveFeatureEliminator(
+            n_features=input_data.n_features,
+            dropout_rate=self.features_dropout_rate,
+        )
 
-        # TODO: need implementation
-        feature_elimination_results = FeatureEliminationResults()
+        feature_elimination_results = {}
 
-        for feature_set in rfe.eliminate_features():
+        for feature_set in rfe.iter_features():
             inner_results = []
+
             for inner_split in data_splitter.iter_inner_splits(outer_split):
                 inner_loop_data = input_data.split_data(inner_split, feature_set)
                 inner_results.append(
                     self.feature_evaluator.evaluate_features(inner_loop_data)
                 )
-            feature_elimination_results.add(inner_results)
 
-        # TODO: we need somewhere here the outer loop evaluation no?
-        return feature_elimination_results
+            feature_elimination_results[feature_set] = inner_results
+            rfe.remove_features(inner_results)
 
-    def get_groups(self, groups, size):
-        if groups is None:
-            logging.info("groups is not specified: i.i.d. samples assumed")
-            groups = np.arange(size)
-        return groups
+        outer_loop_results = self.compute_outer_loop_results(
+            feature_elimination_results, input_data, outer_split
+        )
 
-    def _execute_repetitions(self, outer_loop: OuterLoop) -> List[Repetition]:
-        results = []
-        for _ in range(self.repetitions):
-            outer_loop.refresh_splits()
-            result = outer_loop.run(executor=self.executor)
-            results.append(result)
-        return results
-
-    def _make_outer_loop(self, input_data: InputData) -> OuterLoop:
-        feature_evaluator = self._make_feature_evaluator(input_data)
-        return OuterLoop(
-            self.n_outer,
-            feature_evaluator,
-            self.features_dropout_rate,
-            self.robust_minimum,
+        return outer_loop_results
+
+    def compute_outer_loop_results(self, feature_elimination_results, input_data, outer_split):
+        n_feats_to_score = self._compute_score_curve(feature_elimination_results)
```

Comment:
From alan review of the other PR:
> Aggregation and handling of results although already much improved through the Postprocessor class. Still it seems scattered a bit (e.g. RecursiveFeatureEliminator._compute_score_curve, FeatureEvaluator._get_feature_ranks)

maybe we can move it to the utils/post_processor method as pure functions? or something else

Reply:
sounds good

Reply:
I added it as a TODO comment for now